### PR TITLE
feat(amazon): implement target tracking policy support

### PR DESCRIPTION
**Pull request by anotherchrisberry**
This ended up being kind of gnarly because we are doing some unfortunate things with the server group details in the transformation, manipulating the raw scaling policies into a form that is friendly to the details view.

Otherwise, it's a direct port from our internal implementation of target tracking policies.

### CODE REVIEW

File: app/scripts/modules/amazon/src/loadBalancer/configure/choice/awsLoadBalancerChoice.modal.html [hunk 1]
```diff
@@ -4,8 +4,8 @@
     <h3>Select Type of Load Balancer</h3>
   </div>
   <div class="modal-body">
-    <div class="load-balancer-choices">
-      <div class="load-balancer-choice clickable"
+    <div class="card-choices">
+      <div class="card"
```
Comment (jrsquared) [inline app/scripts/modules/amazon/src/loadBalancer/configure/choice/awsLoadBalancerChoice.modal.html]: nice.

File: app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/CreateScalingPolicyButton.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  }
+
+  public handleClick(): void {
+    this.setState({showSelection: true});
+  }
+
+  public createStepPolicy(): void {
+    const { serverGroup, application } = this.props;
+
+    const policy: IScalingPolicy = {
```
Comment (jrsquared) [inline app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/CreateScalingPolicyButton.tsx]: These default values should probably live somewhere outside of the actual UI control itself, like how `constructNewClassicLoadBalancerTemplate` works.

File: app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/CreateScalingPolicyButton.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        policy: () => policy,
+        serverGroup: () => serverGroup,
+        application: () => application,
+      }
+    });
+  }
+
+  public createTargetTrackingPolicy(): void {
+    const { serverGroup, application } = this.props;
+    const policy: any = {
```
Comment (jrsquared) [inline app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/CreateScalingPolicyButton.tsx]: This one too

File: app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/targetTracking/targetTrackingPopover.html [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<target-tracking-chart style="height: 150px;"
+                       config="$ctrl.policy.targetTrackingConfiguration"
+                       ticks="{x: 12, y: 5}"
```
Comment (jrsquared) [inline app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/targetTracking/targetTrackingPopover.html]: `target-tracking-chart` and `metric-alarm-chart` both use the same style/layout defaults, maybe pull them out?

File: app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/targetTracking/upsertTargetTracking.modal.html [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+        </div>
+
+        <div class="row">
+          <div class="col-md-10 col-md-offset-1">
+            <div style="height: 220px">
+              <target-tracking-chart style="height: 150px;"
+                                     config="$ctrl.command.targetTrackingConfiguration"
+                                     alarm-updated="$ctrl.alarmUpdated"
+                                     ticks="{x: 12, y: 5}"
```
Comment (jrsquared) [inline app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/targetTracking/upsertTargetTracking.modal.html]: Same height/ticks/margins again...

File: app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/targetTracking/targetTrackingChart.component.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    serverGroup: '<',
+    state: '=',
+    alarmUpdated: '<',
+  },
+  controller: TargetTrackingChartController,
+  template: `
+    <metric-alarm-chart alarm="$ctrl.alarm"
+                        style="height: 150px;"
+                        alarm-updated="$ctrl.alarmUpdated"
+                        ticks="{x: 12, y: 5}"
```
Comment (jrsquared) [inline app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/targetTracking/targetTrackingChart.component.ts]: and same again...

File: app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/upsert/PolicyTypeSelectionModal.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    return (
+      <Modal show={true} onHide={this.close}>
+        <Modal.Header closeButton={true}>
+          <h3>Select a policy type</h3>
+        </Modal.Header>
+        <Modal.Body>
+          <div className="card-choices">
+            <div className={targetClass} onClick={this.selectType} id="targetTracking">
+              <h3>Target Tracking</h3>
+              <div>Continuously adjusts the size of the ASG, as the best-effort attempt to keep selected metric at the target value</div>
```
Comment (jrsquared) [inline app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/upsert/PolicyTypeSelectionModal.tsx]: This sentence is awkward. Too sunburned to think of a new one though.

File: app/scripts/modules/core/src/modal/modals.less [hunk 8]
```diff
@@ -319,3 +319,38 @@ abbr.select2-search-choice-close {
     padding: 0 5px;
   }
 }
+
+.card-choices {
```
Comment (jrsquared) [inline app/scripts/modules/core/src/modal/modals.less]: nice

File: app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/CreateScalingPolicyButton.tsx (hunk 2 above)
Comment (anotherchrisberry) [inline app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/CreateScalingPolicyButton.tsx]: yeah, good call, will move

File: app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/targetTracking/targetTrackingPopover.html (hunk 4 above)
Comment (anotherchrisberry) [inline app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/targetTracking/targetTrackingPopover.html]: yep, agreed, will move

File: app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/upsert/PolicyTypeSelectionModal.tsx (hunk 7 above)
Comment (anotherchrisberry) [inline app/scripts/modules/amazon/src/serverGroup/details/scalingPolicy/upsert/PolicyTypeSelectionModal.tsx]: A very nice Russian wrote this, but I will make it more palatable to your American eare.

**Pull request merged**